File: src/tauri/commands/_mock-data.ts
import type { LoginResponse } from "@/models/auth";
import type { MenuConfig } from "@/models/menu-config";
import type { EffectiveMenuPermission } from "@/models/menu-permission";

/** Nhóm menu top-level (không thuộc menu_group nào), khớp UNGROUPED trong menu store. */
const UNGROUPED = "—";

export const mockLoginResponse: LoginResponse = {
  user_id: 1,
  username: "demo",
  full_name: "Nguyễn Minh Thông",
  email: "thongnm@allexceed.co.jp",
  roles: ["admin"],
};

export const mockMenuConfigs: MenuConfig[] = [
  { key: "overview", title: "Overview", path: "/overview", icon: "pi-home", group: UNGROUPED, visible: true, order: 1 },
  { key: "tool-snippets", title: "Snippets", path: "/tools/snippets", icon: "pi-code", group: "Tools", visible: true, order: 10 },
  { key: "tool-converter", title: "Converter", path: "/tools/converter", icon: "pi-sync", group: "Tools", visible: true, order: 11 },
  { key: "cloud-storage", title: "Storage", path: "/cloud/storage", icon: "pi-cloud-upload", group: "Cloud", visible: true, order: 20 },
  { key: "cloud-deploy", title: "Deployments", path: "/cloud/deploy", icon: "pi-server", group: "Cloud", visible: true, order: 21 },
  { key: "ai-chat", title: "Chat Assistant", path: "/ai/chat", icon: "pi-comments", group: "AI Agent", visible: true, order: 30 },
  { key: "gov-users", title: "Users", path: "/governance/users", icon: "pi-users", group: "Governance", visible: true, order: 40 },
  { key: "gov-roles", title: "Roles", path: "/governance/roles", icon: "pi-shield", group: "Governance", visible: true, order: 41 },
  { key: "settings", title: "Settings", path: "/settings", icon: "pi-cog", group: UNGROUPED, visible: true, order: 100 },
];

export const mockEffectivePermissions: EffectiveMenuPermission[] = mockMenuConfigs.map((m) => ({
  menu_key: m.key,
  is_allowed: true,
  role_allowed: true,
  source: "role",
}));

File: src-tauri/src/services/mock_data.rs
//! Dữ liệu mock dùng khi database chưa được cấu hình/kết nối được, để có thể
//! xem layout (login, sidebar, phân quyền) trong lúc phát triển mà không cần
//! Postgres thật. Chỉ được dùng ở debug build — xem `cfg!(debug_assertions)`
//! tại nơi gọi trong các service tương ứng.

use crate::models::auth::LoginResponse;
use crate::models::menu_entity::MenuEntity;
use crate::models::menu_permission::EffectiveMenuPermission;

/// Nhóm menu top-level (không thuộc menu_group nào), khớp `UNGROUPED` trong menu store (frontend).
const UNGROUPED: &str = "—";

pub fn mock_login_response(username: &str) -> LoginResponse {
    LoginResponse {
        user_id: 1,
        username: if username.is_empty() { "demo".to_string() } else { username.to_string() },
        full_name: "Nguyễn Minh Thông".to_string(),
        email: "thongnm@allexceed.co.jp".to_string(),
        roles: vec!["admin".to_string()],
    }
}

pub fn mock_menu_configs() -> Vec<MenuEntity> {
    vec![
        MenuEntity {
            key: "overview".into(),
            title: "Overview".into(),
            path: "/overview".into(),
            icon: "pi-home".into(),
            group: UNGROUPED.into(),
            visible: true,
            order: 1,
        },
        MenuEntity {
            key: "Git".into(),
            title: "Snippets".into(),
            path: "/git".into(),
            icon: "pi-github".into(),
            group: UNGROUPED.into(),
            visible: true,
            order: 10,
        },
        MenuEntity {
            key: "ai-chat".into(),
            title: "Chat Assistant".into(),
            path: "/ai/chat".into(),
            icon: "pi-comments".into(),
            group: UNGROUPED.into(),
            visible: true,
            order: 30,
        },
        MenuEntity {
            key: "gov-users".into(),
            title: "Users".into(),
            path: "/governance/users".into(),
            icon: "pi-users".into(),
            group: "Governance".into(),
            visible: true,
            order: 40,
        },
        MenuEntity {
            key: "gov-roles".into(),
            title: "Roles".into(),
            path: "/governance/roles".into(),
            icon: "pi-shield".into(),
            group: "Governance".into(),
            visible: true,
            order: 41,
        },
        MenuEntity {
            key: "settings".into(),
            title: "Settings".into(),
            path: "/settings".into(),
            icon: "pi-cog".into(),
            group: UNGROUPED.into(),
            visible: true,
            order: 100,
        },
    ]
}

pub fn mock_effective_permissions() -> Vec<EffectiveMenuPermission> {
    mock_menu_configs()
        .into_iter()
        .map(|m| EffectiveMenuPermission {
            menu_key: m.key,
            is_allowed: true,
            role_allowed: true,
            source: "role".into(),
        })
        .collect()
}
